run-integration-test/kustomize/bases/opentelemetry-collectors/kubernetes-objects/opentelemetrycollector.yml

```diff
@@ -0,0 +1,73 @@
+---
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  name: kubernetes-objects
+spec:
+  serviceAccount: kubernetes-objects
+  env: []
+  envFrom: []
+  config:
+    receivers:
+      # https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/k8sobjectsreceiver/testdata/config.yaml
+      k8sobjects:
+        objects:
+        # TODO: Add Stackable objects, and other common k8s objects
+        - name: pods
+          mode: watch
+        - name: events
+          mode: watch
+          group: events.k8s.io
+          exclude_watch_type: [DELETED]
+
+    extensions: {}
+
+    processors:
+      # https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor
+      k8sattributes:
+        # Use the k8s attributes set by the receiver
+        # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor#as-a-gateway
+        passthrough: false
+
+      # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/attributesprocessor
+      # These are attributes for the specific spans, log events, etc...
+      # For higher level attributes see the "resource" attributes below.
+      attributes:
+        actions: []
+
+      # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourceprocessor
+      # These are top level attributes about the application as a whole and not any specific event.
+      # These are useful for correlating. For span/event specific atrtibutes, see above.
+      resource:
+        attributes:
+          # This is to prevent the event label "service_name" having the value "unknown_service"
+          - action: upsert
+            key: service.name
+            value: kubernetes-objects
+
+      resourcedetection/env:
+        detectors: [env]
+        timeout: 2s
+        override: false
+      # yamllint disable rule:comments-indentation
+      # memory_limiter:
+      #   check_interval: 1s
+      #   limit_percentage: 75
+      #   spike_limit_percentage: 15
+      # batch:
+      #   send_batch_size: 10000
+      #   timeout: 10s
+      # yamllint enable rule:comments-indentation
+
+    # Configure exporters in the overlay
+    exporters: {}
+
+    service:
+      extensions: []
+      pipelines:
+        logs:
```

Comment:
question: This is the name of the pipeline, right? As far as I can see the name can be freely chosen. Should we instead name it `kubernetes-objects` instead?

Reply:
`logs` is the _type_ of the pipeline. It can be named specifically though, like `logs/ship-to-loki`.

This makes the overlays a little more complicated, so I opted for making separate collectors (per-intent), instead of complex pipelines.